### ISSUE TITLE
Get pkgsCross.wasi32.buildPackages.haskell.compiler.{ghc961,ghcHEAD} to work

### ISSUE THREAD

**Issue by sternenseemann** [NixOS/nixpkgs#220835]
Some hints can be gathered from [ghc-wasm-meta](https://gitlab.haskell.org/ghc/ghc-wasm-meta). First prerequisite seems to be packaging `libffi-wasi`.

**Comment by sternenseemann** [comment]
Some progress in #225000, but we'll need to somehow use LLVM >= 16 to compile GHC. As it turns out GHC upstream also has a custom fork of wasi-sdk which I hope we can avoid packaging.

**Comment by sternenseemann** [comment]
I've put https://github.com/NixOS/nixpkgs/pull/225000 on hold, since GHC requires an unreasonable amount of patches to its dependencies for the backend to work. The preparatory wasilibc update is at #241739, currently waiting on the failing `firefox-esr` to leave the rotation.

**Comment by danwdart** [comment]
Does this look like it'll be possible using standard nixpkgs? I did check the PR but it seems to be several versions out of date.
